tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/ElasticsearchCMISTests.java

```diff
@@ -0,0 +1,138 @@
+import org.alfresco.utility.data.DataUser;
+import org.alfresco.utility.model.FileModel;
+import org.alfresco.utility.model.FileType;
+import org.alfresco.utility.model.SiteModel;
+import org.alfresco.utility.model.TestGroup;
+import org.alfresco.utility.model.UserModel;
+import org.alfresco.utility.network.ServerHealth;
+import org.alfresco.utility.testrail.ExecutionType;
+import org.alfresco.utility.testrail.annotation.TestRail;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+@ContextConfiguration(locations = "classpath:alfresco-elasticsearch-context.xml",
+                      initializers = AlfrescoStackInitializer.class)
+/**
+ * In this test we are verifying end-to-end the indexing and CMIS queries against Elasticsearch.
+ * In order to test ACLs we created 2 sites and 3 users.
+ */
+public class ElasticsearchCMISTests extends AbstractTestNGSpringContextTests
+{
+    private static final String PREFIX = getAlphabeticUUID() + "_";
+    private static final String UNIQUE_WORD = getAlphabeticUUID();
+    private static final String FILE_0_NAME = PREFIX + "test.txt";
+    private static final String FILE_1_NAME = PREFIX + "another.txt";
+    private static final String FILE_2_NAME = PREFIX + "user1Old.txt";
+    /** This is a file that user 1 doesn't have access to and so shouldn't be returned in their search results. */
+    private static final String USER_2_FILE_NAME = PREFIX + "user2.txt";
+
+    @Autowired
+    private DataUser dataUser;
+
+    @Autowired
+    private DataContent dataContent;
+
+    @Autowired
+    private DataSite dataSite;
+
+    @Autowired
+    private ServerHealth serverHealth;
+
+    @Autowired
+    protected SearchQueryService searchQueryService;
+
+    private UserModel user1;
+    private UserModel user2;
+    private UserModel userMultiSite;
+    private SiteModel siteModel1;
+    private SiteModel siteModel2;
+
+    /**
+     * Data will be prepared using the schema below:
+     * <p>
+     * Site1:
+     * - Users: user1, userMultiSite
+     * - Documents: FILE_0_NAME (owner: user1), FILE_1_NAME (owner: user1), FILE_2_NAME (owner: user2)
+     * <p>
+     * Site2:
+     * - Users: user2, userMultiSite
+     * - Documents: USER_2_FILE_NAME (owner: user2)
+     */
+    @BeforeClass(alwaysRun = true)
+    public void dataPreparation()
+    {
+        serverHealth.assertServerIsOnline();
+
+        user1 = dataUser.createRandomTestUser();
+        user2 = dataUser.createRandomTestUser();
+        userMultiSite = dataUser.createRandomTestUser();
+
+        siteModel1 = dataSite.usingUser(user1).createPrivateRandomSite();
+        siteModel2 = dataSite.usingUser(user2).createPrivateRandomSite();
+
+        dataUser.addUserToSite(user2, siteModel1, UserRole.SiteContributor);
```

Comment:
I might be missing something, but shouldn't user2 only be added to Site2 and user1 to Site1?

Reply:
I borrowed most of this data set up from another test because it seemed to be what I wanted too. User2 owns one document in each site, but later they are removed from site1.  We should add an ACL test too - I'll create a task for this.